lib/generators.nix

```diff
@@ -14,29 +14,81 @@
  * Documentation in the manual, #sec-generators
  */
 { lib }:
-with (lib).trivial;
-let
-  libStr = lib.strings;
-  libAttr = lib.attrsets;
-
-  inherit (lib) isFunction;
-in
 
-rec {
+let
+  inherit (lib)
+    addErrorContext
+    assertMsg
+    attrNames
```

Comment:
What's the reason behind using `lib` only for some fields but `lib.<sublib>` for others? E.g. `attrNames` comes from `lib`, but `isDerivation` comes from `lib.attrsets`, even though they're available in both.

Reply:
This was the first PR I made against `lib`, so the _historical_ reason is that I was trying to do the work by hand and I didn't have the script I later wrote to gather and use the names automatically.

If you have a preference, I'd be glad to accommodate.

Reply:
Discussed [on Matrix](https://matrix.to/#/!djTaTBQyWEPRQxrPTb:nixos.org/$jWTFrysx9Y8swFg5ejBb_ba2eVpoFgwVAgb5YesCcqI?via=nixos.org&via=matrix.org&via=nixos.dev), let's use the ones from `lib`. While I did do it the other way around in [`lib.fileset`](https://github.com/NixOS/nixpkgs/blob/master/lib/fileset/default.nix#L101), I think it should just be consistent within the same file for now.

Reply:
Done.